lib/datadog/statsd.rb

```diff
@@ -494,5 +531,11 @@ def send_to_socket(message)
       self.class.logger.error { "Statsd: #{boom.class} #{boom}" } if self.class.logger
       nil
     end
+
+    ##
+    # Use a monitor to synchronize the sending of a message to the socket
+    def send_to_socket_synchronized(message)
```

Comment:
put the synchronize into send_to_socket ?

Reply:
That's what I did first, but since the `send_to_socket` method is huge already, I decided to use a separate wrapper - but I'm fine with either one. `send_to_socket` is way too complex anyway and should be refactored - but that's a different story.